kusto/ingest/ingest.go

```diff
@@ -64,12 +64,17 @@ type Ingestion struct {
 
 // New is the constructor for Ingestion.
 func New(client *kusto.Client, db, table string) (*Ingestion, error) {
+	return NewWithBlobProperties(client, db, table, filesystem.BlockSize, filesystem.Concurrency)
```

Comment:
We shouldn't do this or expose a NewWithBlobProperties type.  

We should simply add options.

```go
func New(client *kusto.Client, db, table string, options ...Option) (*Ingestion, error) {
  i := &Ingestion{...}
  for _, o := range options  {
    o(i)
  }
  ...
}
```
Please reach out to me if this isn't clear.

Reply:
I now added the options pattern.
It got a bit messy because I needed to add it to both `filesystem` and `ingest`,
and also because NewSyncPool needs the arguments at the ctor.

What do you think? (also @yogilad )